### PR TITLE
DFTHG # 989 ERA and Road Test - ICBC Re-exams Document Requirements Expiring Based on DPSed Processing Date instead of Calendar days

### PR DESCRIPTION
**Pull request by shruthi-rapala**
# Description

This PR includes the following proposed change(s):

DFTHG # 989 ERA and Road Test - ICBC Re-exams Document Requirements Expiring Based on DPSed Processing Date instead of Calendar days
- Fix the dps processing date for ERA results
- Updated logic to check both the paths for dps processing date calculation

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

### Screenshots

Include screen captures or recordings with captions of the tests run (unit tests, postman request/response and or manual UI driven tests.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the [style guidelines](https://justice.gov.bc.ca/wiki/display/JAM/RCC+Style+Guide) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs [hunk 1]
```diff
@@ -4262,12 +4264,15 @@ public DateTimeOffset GetDpsUplaodDateForERAResults()
                       && x.dfp_dpsclass.Contains("1 - Priority")
                         && (x._dfp_documenttypeid_value == eraResultsGuid || x._dfp_documenttypeid_value == reExamResultsGuid)
                         )
-                    .OrderByDescending(i => i.dfp_uploadeddate)
```
Comment (jburditt) [inline cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4265]: good move removing Take(1) but did you intend to remove the sort? Seems useful but I'm not familiar with requirements, marking this resolved

Comment (jburditt) [reply cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4265]: wait a minute, am I crazy or will this method always return today? It effectively does nothing. What is the intended purpose of this method? To be fair, I don't think it did anything before you made changes to it.

Comment (GeorgeWalker) [reply cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4265]: @jburditt  - this code does not look correct.  @shruthi-rapala - it needs to return result, and you also need the OrderByDescending.

File: cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs [hunk 2]
```diff
@@ -4283,17 +4288,28 @@ public async Task UpdateNonComplyDocuments()
...
+            var dpsUploadDateEraResults = GetDpsUploadDateForERAResults();
+
+            //Path 1 : This is non comply for all the documents except ERA results and Road Test
+
+            UpdateNonComplyDocumentsQuery(dpsUploadedDate);
+
+            // Path 2 : This is to set non comply on  ERA results and Road Test
 
 
+            UpdateNonComplyDocumentsQuery(dpsUploadDateEraResults);
```
Comment (jburditt) [inline cms-adapter/src/Rsbc.Dmf.CaseManagement/Manager/CaseManager.cs:4300]: if I understand this correctly, this method changes the submittal status from all documents open required to 100000005 below the parameter date. If that is true, you could do the same thing by combining to one method call:
```
var maxDate = DateTime.Max(dpsUploadedDate, dpsUploadDateEraResults);
UpdateNonComplyDocumentsQuery(maxDate);
```

**Review by jburditt — APPROVED** [review]

**Pull request merged**